gbfs-validator/__test__/cli.test.js

```diff
@@ -1,57 +1,69 @@
+let path = require('path')
+let exec = require('node:child_process').exec
+
 const serverOpts = {
   port: 0,
   host: '127.0.0.1',
 }
 
-describe('cli', () => {
-  describe('without arguments', () => {
-    test('should show help without required url', async () => {
-      const mockExit = jest.spyOn(process, 'exit').mockImplementation(() => {})
-      const mockConsoleError = jest
-        .spyOn(console, 'error')
-        .mockImplementation(() => {})
-
-      expect(() => {
-        require('../cli.js')
-      }).toThrow('Missing URL')
-
-      expect(mockExit).toHaveBeenCalledWith(1)
-    })
+const cliExec = (args) => {
+  const command = `${path.resolve(`${__dirname}/../cli.js`)} ${args.join(' ')}`
+  return new Promise(resolve => {
+    exec(command,
+      // Setting exit override to allow program to exit simplifying unit testing
+      { env: { ...process.env, 'EXIT_OVERRIDE': 'true' } },
+      (error, stdout, stderr) => {
+        resolve({
+          code: error && error.code ? error.code : 0,
+          error,
+          stdout,
+          stderr
+        })
+      })
   })
+}
 
-  describe('with arguments', () => {
-    let gbfsFeedServer
+describe('cli', () => {
 
-    beforeAll(async () => {
-      gbfsFeedServer = require('./fixtures/server')()
+  let gbfsFeedServer
+  let feedUrl
 
-      await gbfsFeedServer.listen(serverOpts)
+  beforeAll(async () => {
+    gbfsFeedServer = require('./fixtures/server')()
+    await gbfsFeedServer.listen(serverOpts)
+    feedUrl = `http://${gbfsFeedServer.server.address().address}:${gbfsFeedServer.server.address().port}/gbfs.json`
+  })
 
-      return gbfsFeedServer
-    })
+  afterAll(() => {
+    return gbfsFeedServer.close()
+  })
 
-    afterAll(() => {
-      return gbfsFeedServer.close()
-    })
+  test('should show an error if url parameter is not set', async () => {
+    const result = await cliExec([])
+    expect(result.code).toBe(1)
+    expect(result.error.message).toContain('error: required option \'-u, --url <feed_url>\' not specified')
+  })
 
-    test('should run cli', async () => {
-      const url = `http://${gbfsFeedServer.server.address().address}:${
-        gbfsFeedServer.server.address().port
-      }`
+  test('should success and print the report when url parameter set and -pr is set as default', async () => {
```

Comment:
(Only if you accept the suggestion to use `-p` for consistency with `-s`)

```suggestion
  test('should success and print the report when url parameter set and -p is set as default', async () => {
```